Resources/public/js/views/services/plugins/ez-userdraftsplugin.js

```diff
@@ -0,0 +1,245 @@
+                    contentService.loadContentInfo(contentId, Y.bind(function (error, response) {
+                        if (error) {
+                            reject('Cannot load content info: ' + contentId);
+                            this._target.set('loadingError', true);
+                            this._resetTempVariables();
+
+                            return;
+                        }
+
+                        resolve(response.document.Content);
+                    }, this));
+                }, this)));
+            });
+
+            return Y.Promise.all(promises);
+        },
+
+        /**
+         * Loads content types for each content version
+         *
+         * @method _loadDraftContentType
+         * @protected
+         * @param data {Array} list of content
+         * @return {Y.Promise}
+         */
+        _loadDraftContentType: function (data) {
+            var contentTypeService = this.get('host').get('capi').getContentTypeService(),
+                drafts = this.get('drafts'),
+                promises = [];
+
+            data.forEach(Y.bind(function (contentInfo, index) {
+                drafts.item(index).set('contentInfo', contentInfo);
+
+                promises.push(new Y.Promise(Y.bind(function (resolve, reject) {
+                    contentTypeService.loadContentType(contentInfo.ContentType._href, Y.bind(function (error, response) {
+                        if (error) {
+                            reject('Cannot load content type data: ' + contentInfo.ContentType._href);
+                            this._target.set('loadingError', true);
+                            this._resetTempVariables();
+
+                            return;
+                        }
+
+                        resolve(response.document.ContentType);
+                    }, this));
+                }, this)));
+            }, this));
+
+            return Y.Promise.all(promises);
+        },
+
+        /**
+         * Creates a content type model, fills with data and assigns it to each draft
+         *
+         * @method _fillDraftContentTypeModel
+         * @protected
+         * @param data {Array} list of content types
+         */
+        _fillDraftContentTypeModel: function (data) {
+            var drafts = this.get('drafts');
+
+            drafts.each(function (version, index) {
+                var contentType = new Y.eZ.ContentType();
+
+                contentType.loadFromHash(data[index]);
+
+                version.set('contentType', contentType);
+            }).sort();
+
+            this._target.set(this._targetAttributeName, drafts);
+            this._resetTempVariables();
+        },
+    }, {
+        NS: 'userDrafts',
+        ATTRS: {
+            /**
+             * The drafts list
+             *
+             * @attribute drafts
+             * @type Y.ModelList
+             */
+            drafts: {
+                valueFn: function () {
+                    return new Y.ModelList({
```

Comment:
we can not use a ModelList because what you'll return to the view is not just a model list and a list of Object containing a VersionInfo model, a ContentInfo model and a ContentType model